src/utils/ViewMapLeaf.ts

```diff
@@ -0,0 +1,96 @@
+export class ViewMapLeaf<T> {
+  public containerType?: ContainerType;
+  public parent: ViewMapLeaf<T>;
+  public childs: ViewMapLeaf<T>[];
+
+  private value?: T;
+
+  constructor () {
+    this.childs = [];
+  }
+
+  public isContainer (): boolean {
+    return this.getValue() === undefined;
+  }
```

Comment:
It seems that you've tried to combine both a branch and a leaf in a single class. Splitting them into `Branch` and `Leaf` would bring a couple of benefits:

* The name wouldn't be misleading: calling a branch a leaf is at least confusing.
* You would avoid optional properties
* You could have used polymorphism instead of checking `isContainer`.